http.c

```diff
@@ -150,7 +153,7 @@ int http_open(struct http_get *hg, int timeout_ms)
 	char *proxy = getenv("http_proxy");
 	if (proxy) {
 		hg->proxy = xnew(struct http_uri, 1);
-		if (http_parse_uri(proxy, hg->proxy)) {
+		if (parse_uri(proxy, hg->proxy)) {
```

Comment:
Trivial. Below: `Failed to parse HTTP proxy URI` becomes `Failed to parse HTTP(s) proxy URI`.

Reply:
Good catch, I hadn't considered the possibility that http_proxy could contain both HTTP/HTTPS URIs. I realize my assumption about `http_proxy` and `https_proxy` env variables were wrong and the current behavior is not right.

So before changing the error message we'll need to do a few things :
- Add a "not supported" warning message for `https_proxy` env variable (not supported until we implement CONNECT requests) and remove the current error message for HTTP proxy / HTTPS target URI which does not make sense.
- Test HTTPS proxy for HTTP target URIs (and if it works fine, add the previous error message).

I think this should be enough to remove any surprising behavior.